OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -1299,13 +1300,16 @@ public static void setEmail(@NonNull final String email, @Nullable final String
       if (taskController.shouldQueueTaskForInit(OSTaskController.SET_EMAIL)) {
          logger.error("Waiting for remote params. " +
                  "Moving " + OSTaskController.SET_EMAIL + " operation to a pending task queue.");
+         if (callback != null)
+            callback.onTaskQueued();
```

Comment:
I don't think we should change / add new APIs to address this bug fix PR. However I am open to a proposal on a new API to best handle this scenario for all our methods for consistency.

Just some feedback on the API itself though, I would expect a return value from `setEmail` instead of a callback that it has been queued.

Reply:
Noted, I will keep this in mind for a future integration